library/src/main/java/com/wultra/android/powerauth/networking/utils/AppUtils.kt

```diff
@@ -34,4 +37,60 @@ class AppUtils {
             return appContext.packageManager.getPackageInfo(appContext.packageName, flags)
         }
     }
-}
\ No newline at end of file
+}
+
+object ConnectionMonitor {
+
+    enum class Status(val value: String) {
+        UNKNOWN("unknown"),
+        NO_CONNECTION("noConnection"),
+        WIFI("wifi"),
+        CELLULAR("cellular"),
+        WIRED("wired")
+    }
+
+    private var status: Status = Status.NO_CONNECTION
```

Comment:
Is the property ever really utilized? It seems the values is always locally (in `getConnectivityStatus()`) resolved and the "cached" value is never used. It could be just local variable.